components/clp-package-utils/clp_package_utils/scripts/decompress.py

```diff
@@ -185,7 +185,7 @@ def handle_extract_stream_cmd(
         logger.error(f"JSON extraction is not supported for storage engine `{storage_engine}`.")
         return -1
 
-    dataset = parsed_args.dataset
+    dataset = getattr(parsed_args, "dataset", None)
```

Comment:
Perhaps to improve readability and reduce errors like this in the future, we can move any IR-specific stuff into the IR block on line 220. And similarly for the JSON block.